clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -737,8 +750,8 @@ def _reconnect(self):
             except TransportInitError:
                 self.socket = None
                 
-            if self.socket is None:
-                # exponential backoff
+            if self.socket is None and interval < 30.:
```

Comment:
nitpick: I know that `30.` is still technically correct to force it as a float literal, but `30.0` is the style used elsewhere, so I'd appreciate updating this for consistency.